### PR TITLE
Completed the iDEAL flow

### PR DESCRIPTION
**Pull request by mpetrenco**
### Features:
- Integrated complete iDEAL payment flow with `api.judopay.com`;
- Added account holder name to the iDEAL request;

### Improvements:
- Limited the iDEAL currency to `EUR` as per requirements;
- Changed iDEAL amount to `double` as per requirements;
- Redone the REST methods to accept the full path and not the endpoint due to multiple origins;
- `merchantPaymentReference` has been trimmed to 40 characters as per requirements;

### Fixes:
- Fixed a small issue regarding polling not stoping on Back button press;
- Fixed wrong localization keys for the iDeal transaction status;

### CODE REVIEW

File: Source/API/JPSession.h [hunk 1]
```diff
@@ -38,6 +38,11 @@ typedef void (^JudoCompletionBlock)(JPResponse *_Nullable, NSError *_Nullable);
  */
 @property (nonatomic, strong, readonly) NSString *_Nonnull endpoint;
 
+/**
+ *  The endpoint for REST API calls to the judo API related to iDEAL transactions
+ */
+@property (nonatomic, strong, readonly) NSString *_Nonnull iDealEndpoint;
```
Comment (stpopa) [inline Source/API/JPSession.h]: I think `https://api.judopay.com/` it's a temporary solution, we may not need this to be exposed

File: Source/Controller/JudoPaymentMethodsViewController.m [hunk 2]
```diff
@@ -247,7 +248,7 @@ - (void)onIDEALButtonTap:(id)sender {
     __weak JudoPaymentMethodsViewController *weakSelf = self;
 
     [self.judoKitSession invokeIDEALPaymentWithJudoId:self.viewModel.judoId
-                                               amount:self.viewModel.amount
+                                               amount:self.viewModel.amount.amount.doubleValue
```
Comment (stpopa) [inline Source/Controller/JudoPaymentMethodsViewController.m]: `.amount.amount.` :)

File: Source/API/JPSession.h (hunk 1 above)
Comment (mpetrenco) [reply Source/API/JPSession.h]: Will remove it once backend stuff is sorted, `api.judopay.com` is probably going to be used for PBBA as well, so I might as well keep it for now.

File: Source/Controller/JudoPaymentMethodsViewController.m (hunk 2 above)
Comment (mpetrenco) [reply Source/Controller/JudoPaymentMethodsViewController.m]: Yea, I made a note on that. Will be changed in a future PR.

**Pull request merged**